test/update-root-package-json.js

```diff
@@ -1,297 +0,0 @@
-      },
-    },
-    'should write new package.json with tree data'
-  )
-})
-
-t.test('existing package.json', async t => {
-  const path = t.testdir({
-    'package.json': JSON.stringify({
-      name: 'existing-package-json-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: {
-        abbrev: '^1.0.0',
-      },
-    }),
-  })
-  await updateRootPackageJson({
-    path,
-    package: {
-      name: 'missing-package-json-test',
-      version: '1.0.0',
-      dependencies: {},
-    },
-  })
-  t.match(
-    require(resolve(path, 'package.json')),
-    {
-      name: 'existing-package-json-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: undefined,
-    },
-    'should write new package.json with tree data'
-  )
-})
-
-t.test('unchanged package.json', async t => {
-  const path = t.testdir({
-    'package.json': JSON.stringify({
-      name: 'existing-package-json-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: {
-        abbrev: '^1.0.0',
-      },
-    }, null, 2) + '\n',
-  })
-  const { mtime } = statSync(path + '/package.json')
-  await updateRootPackageJson({
-    path,
-    package: {
-      name: 'existing-package-json-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: {
-        abbrev: '^1.0.0',
-      },
-    },
-  })
-  t.match(
-    require(resolve(path, 'package.json')),
-    {
-      name: 'existing-package-json-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: {
-        abbrev: '^1.0.0',
-      },
-    },
-    'should write new package.json with tree data'
-  )
-  const { mtime: newMtime } = statSync(path + '/package.json')
-  t.equal(newMtime.toISOString(), mtime.toISOString(), 'mtime not changed')
-})
-
-t.test('existing package.json with optionalDependencies', async t => {
-  const path = t.testdir({
-    'package.json': JSON.stringify({
-      name: 'existing-package-json-optional-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: {},
-    }),
-  })
-  await updateRootPackageJson({
-    path,
-    package: {
-      name: 'missing-package-json-optional-test',
-      version: '1.0.0',
-      dependencies: {
-        abbrev: '^1.0.0',
-      },
-      optionalDependencies: {
-        abbrev: '^1.0.0',
-      },
-    },
-  })
-  t.match(
-    require(resolve(path, 'package.json')),
-    {
-      name: 'existing-package-json-optional-test',
-      version: '1.0.0',
-      bin: './file.js',
-      funding: 'http://example.com',
-      dependencies: undefined,
-      optionalDependencies: {
-        abbrev: '^1.0.0',
-      },
-    },
-    'should write new package.json with tree data'
-  )
-})
-
-t.test('custom formatting', async t => {
```

Comment:
we don't have a test for this in `@npmcli/package-json`, feels like we should add one. i see the code that should do this, but it's always a good idea to prove it works as intended

Reply:
we actually have! 😅 I added it as part of https://github.com/npm/package-json/blob/main/test/index.js#L50 which is easy to see in the snapshot: https://github.com/npm/package-json/blob/main/tap-snapshots/test/index.js.test.cjs#L12 but I see how confusing that is from a maintenance point of view! 😬 

I'll swap that test to just use regular formatting and move that check to a `custom formatting` separated test! thanks for bringing that up!